daemon/src/maker_inc_connections_actor.rs

```diff
@@ -1,95 +1,225 @@
-                    let (out_message_actor, out_message_actor_inbox) = send_wire_message_actor::new(write);
-
-                    tokio::spawn(in_taker_actor);
-                    tokio::spawn(out_message_actor);
-
-                    cfd_maker_actor_inbox.send(maker_cfd_actor::Command::NewTakerOnline{id : taker_id}).unwrap();
-
-                    write_connections.insert(taker_id, out_message_actor_inbox);
-                },
-                Some(message) = our_inbox.recv() => {
-                    match message {
-                        maker_inc_connections_actor::Command::NotifyInvalidOrderId { id, taker_id } => {
-                            let conn = write_connections.get(&taker_id).expect("no connection to taker_id");
-                            conn.send(wire::MakerToTaker::InvalidOrderId(id)).unwrap();
-                        },
-                        maker_inc_connections_actor::Command::BroadcastOrder(order) => {
-                            for conn in write_connections.values() {
-                                conn.send(wire::MakerToTaker::CurrentOrder(order.clone())).unwrap();
-                            }
-                        },
-                        maker_inc_connections_actor::Command::SendOrder {order, taker_id} => {
-                            let conn = write_connections.get(&taker_id).expect("no connection to taker_id");
-                            conn.send(wire::MakerToTaker::CurrentOrder(order)).unwrap();
-                        },
-                        maker_inc_connections_actor::Command::NotifyOrderAccepted { id, taker_id } => {
-                            let conn = write_connections.get(&taker_id).expect("no connection to taker_id");
-                            conn.send(wire::MakerToTaker::ConfirmTakeOrder(id)).unwrap();
-                        },
-                        maker_inc_connections_actor::Command::OutProtocolMsg { taker_id, msg } => {
-                            let conn = write_connections.get(&taker_id).expect("no connection to taker_id");
-                            conn.send(wire::MakerToTaker::Protocol(msg)).unwrap();
-                        }
-                    }
+pub struct BroadcastOrder(pub Option<Order>);
+impl Message for BroadcastOrder {
+    type Result = anyhow::Result<()>;
+}
+
+pub struct SendOrder {
+    pub order: Option<Order>,
+    pub taker_id: TakerId,
+}
+impl Message for SendOrder {
+    type Result = anyhow::Result<()>;
+}
+
+pub struct NotifyInvalidOrderId {
+    pub id: OrderId,
+    pub taker_id: TakerId,
+}
+impl Message for NotifyInvalidOrderId {
+    type Result = anyhow::Result<()>;
+}
+
+pub struct NotifyOrderAccepted {
+    pub id: OrderId,
+    pub taker_id: TakerId,
+}
+impl Message for NotifyOrderAccepted {
+    type Result = anyhow::Result<()>;
+}
+
+pub struct OutProtocolMsg {
+    pub taker_id: TakerId,
+    pub msg: SetupMsg,
+}
+impl Message for OutProtocolMsg {
+    type Result = anyhow::Result<()>;
+}
```

Comment:
I reckon these should be grouped into one enum for sending to a specific taker and one for sending to all. That is way too much boilerplate :D

Reply:
I ended up having 3 different messages: `BroadcastOrder` (sending to all), `TakerMessage` (sending to one taker, with enum inside), and `OutProtocolMsg`, as it's unrelated to both.